substrate/frame/assets/src/types.rs

```diff
@@ -232,6 +232,32 @@ impl<AssetId, AccountId, Balance> FrozenBalance<AssetId, AccountId, Balance> for
 	fn died(_: AssetId, _: &AccountId) {}
 }
 
+/// Trait for specifying a balance that is distinct from the free balance.
+/// This balance is then summed up with the balance of the account, and the
+/// `minimum_balance` (and frozen balance, if any) of the asset to calculate
+/// the reducible balance.
+pub trait BalanceOnHold<AssetId, AccountId, Balance> {
+	/// Return the held balance.
+	///
+	/// If `Some`, it means some balance is suspended, and it can be infallibly
+	/// slashed.
+	///
+	/// If `None` is returned, then nothing special is enforced.
```

Comment:
This should be documented from the perspective of the assets pallet, asset pallet has noting to do with slashing. 
In functions like `reducible_balance` you do assume that `account.balance` does not contain held balance.

Reply:
Agreed. Will update the documentation to reflect these changes. Slashing -> Burning. Including the mention of the difference between `account.balance` and balance _on hold_.